### ISSUE TITLE
Crash in ProxyConfigServiceTor

### ISSUE THREAD

**Issue by iefremov** [brave/brave-browser#12798]
```
[ 00 ] <name omitted>
[ 01 ] net::ProxyConfigServiceTor::SetProxyAuthorization(net::ProxyConfigWithAnnotation const&, GURL const&, net::ProxyResolutionService*, net::ProxyInfo*)
[ 02 ] net::ConfiguredProxyResolutionService::TryToCompleteSynchronously(GURL const&, net::ProxyInfo*)
[ 03 ] net::ConfiguredProxyResolutionService::ResolveProxy(GURL const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, net::NetworkIsolationKey const&, net::ProxyInfo*, base::OnceCallback<void (int)>, std::__1::unique_ptr<net::ProxyResolutionRequest, std::__1::default_delete<net::ProxyResolutionRequest> >*, net::NetLogWithSource const&)
[ 04 ] <name omitted>
[ 05 ] network::NetworkContext::LookUpProxyForURL(GURL const&, net::NetworkIsolationKey const&, mojo::PendingRemote<network::mojom::ProxyLookupClient>)
```
https://brave.sp.backtrace.io/p/brave/explore?time=all&filters=((callstack%2Ccontains%2Cnet%3A%3AProxyConfigServiceTor%3A%3ASetProxyAuthorization))&aggregations=((ver%2Cmax)%2C(callstack%2Chead))&

**Comment by darkdh** [comment]
This should be resolved by https://github.com/brave/brave-core/pull/7069/commits/834a46b8630eb699d678e7869d521504efd356cc

**Comment by iefremov** [comment]
Marked as resolved in backtrace

**Comment by bsclifton** [comment]
Added to `1.19.x` milestone as https://github.com/brave/brave-core/pull/7069 was merged to 1.19.x 😄
